Flat/interface/TemplatedPandaAnalyzer.h

```diff
@@ -177,12 +176,22 @@ void PandaAnalyzer::FillGenTree(panda::Collection<T>& genParticles)
 
   // cluster the  jet 
   fastjet::ClusterSequenceArea seq(finalStates, *jetDef, *areaDef);
-  std::vector<fastjet::PseudoJet> allJets(fastjet::sorted_by_pt(seq.inclusive_jets(0.)));
-
+  allGenJets=fastjet::sorted_by_pt(seq.inclusive_jets(0.));
+  tr->TriggerEvent("Cluster gen jets");
+}
 
+template <typename T>
+void PandaAnalyzer::GenJetProperties(panda::Collection<T>& genParticles, panda::FatJet *matchTarget)
+{
+  // If matchTarget is a valid pointer, GenJetProperties will find the hardest gen fatjet which
+  // (a) does not overlap with a hard lepton from a t->W(lv) decay, and (b) is successfully
+  // matched to the reco fatjet, by having delta-R less than the cone size pt's compatible within 50%.
+  // Otherwise, GenJetProperties will just find the hardest gen fatjet which does not overlap with said hard leptons.
+  // In either case, if no gen fatjet is so found having pT above 200 GeV, nothing is stored.
   fastjet::PseudoJet* fullJet = NULL;
-  for (auto& testJet : allJets) {
-    if (testJet.perp() < 450)
+  float dR2 = FATJETMATCHDR2;
+  for (auto& testJet : allGenJets) {
+    if (testJet.perp() < 200)
```

Comment:
Yeah this needs to be configurable. No way I want things going down to 200 for my studies.